test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_info.cpp

```diff
@@ -0,0 +1,497 @@
+
+        size_t test_dimensions;
+
+        error = clGetMutableCommandInfoKHR(
+            command, CL_MUTABLE_DISPATCH_DIMENSIONS_KHR,
+            sizeof(test_dimensions), &test_dimensions, nullptr);
+        test_error(error, "clGetMutableCommandInfoKHR failed");
+
+        if (test_dimensions != dimensions)
+        {
+            log_error("ERROR: Incorrect command buffer returned from "
+                      "clGetMutableCommandInfoKHR.");
+            return TEST_FAIL;
+        }
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        return CL_SUCCESS;
+    }
+
+    cl_mutable_command_khr command = nullptr;
+    const size_t dimensions = 3;
+};
+
+struct InfoType : public BasicMutableCommandBufferTest
+{
+    using BasicMutableCommandBufferTest::BasicMutableCommandBufferTest;
+
+    InfoType(cl_device_id device, cl_context context, cl_command_queue queue)
+        : BasicMutableCommandBufferTest(device, context, queue)
+    {}
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr,
+            &global_work_size, nullptr, 0, nullptr, nullptr, &command);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        cl_command_type type = 0;
+        error = clGetMutableCommandInfoKHR(command,
+                                           CL_MUTABLE_COMMAND_COMMAND_TYPE_KHR,
+                                           sizeof(type), &type, NULL);
+        test_error(error, "clGetMutableCommandInfoKHR failed");
+
+        if (type != CL_COMMAND_NDRANGE_KERNEL)
+        {
+            log_error("ERROR: Wrong type returned from "
+                      "clGetMutableCommandInfoKHR.");
+            return TEST_FAIL;
+        }
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        return CL_SUCCESS;
+    }
+
+    cl_mutable_command_khr command = nullptr;
+};
+
+struct InfoQueue : public BasicMutableCommandBufferTest
+{
+    using BasicMutableCommandBufferTest::BasicMutableCommandBufferTest;
+
+    InfoQueue(cl_device_id device, cl_context context, cl_command_queue queue)
+        : BasicMutableCommandBufferTest(device, context, queue)
+    {}
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr,
+            &global_work_size, nullptr, 0, nullptr, nullptr, &command);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        cl_command_queue testQueue = nullptr;
+        error = clGetMutableCommandInfoKHR(
+            command, CL_MUTABLE_COMMAND_COMMAND_QUEUE_KHR, sizeof(testQueue),
+            &testQueue, nullptr);
+        test_error(error, "clGetMutableCommandInfoKHR failed");
+
+        if (!testQueue)
```

Comment:
I think we could actually check `queue == testQueue` here, see https://github.com/KhronosGroup/OpenCL-CTS/pull/1611#discussion_r1113613069

This also applies to `kernel = test_kernel`

Reply:
For `test_queue == queue` it works, but for `test_kernel == kernel` it fails (diffrent addresses).

Reply:
Interesting, `test_kernel == kernel` passes on Codeplay's implementation. Which implementation are you using? We can tag the developer to see if that's just a bug or if there's an issue with specifying a query for `CL_MUTABLE_DISPATCH_KERNEL_KHR`

Reply:
I am using implementation from: https://github.com/bashbaug/SimpleOpenCLSamples/tree/main/layers/10_cmdbufemu

Reply:
cool, @bashbaug do you have thoughts on testing for equality between the `cl_kernel` returned by `CL_MUTABLE_DISPATCH_KERNEL_KHR` and the `cl_kernel` passed on `clCommandNDRangeKernelKHR` ?